pygmt/datasets/earth_relief.py

```diff
@@ -36,7 +42,21 @@ def load_earth_relief(resolution="01d"):
 
     """
     _is_valid_resolution(resolution)
-    fname = which("@earth_relief_{}".format(resolution), download="u")
+
+    if registration in ("pixel", "gridline", None):
+        _registration = ""  # If None, let GMT decide on Pixel/Gridline type
+        with clib.Session() as lib:
+            if registration is not None and Version(lib.info["version"]) >= Version(
+                "6.1.0"
+            ):
```

Comment:
I think this is also good:
```suggestion
            if registration and Version(lib.info["version"]) >= Version("6.1.0"):
```